rspace/src/test/scala/coop/rchain/rspace/StorageActionsTests.scala

```diff
@@ -268,6 +289,9 @@ trait StorageActionsTests extends StorageTestsBase[String, Pattern, String, Stri
 
     store.eventsCounter.getProducesCount shouldBe 2
     store.eventsCounter.getConsumesCount shouldBe 1
+    store.eventsCounter.getProducesCommCount shouldBe 1
+    store.eventsCounter.getConsumesCommCount shouldBe 0
```

Comment:
I'm trying to understand how does _produce COMM event_ differs from _consume COMM event_. In the previous test you have: _producing and then consuming on the same channel_ which results in _getConsumesCommCount shouldBe 1_; here we have _producing on channel, consuming on that channel ... _ but _getConsumesCommCount == shouldBe 0 _. I don't understand that, the scenario is _almost_ the same as in the previous test case. Why is it that now there is only _produce COMM event_ ?

Reply:
Difference is in order of produce and consume calls. If we call consume before produce, than produce comm even occur. If we call produce and then consume - this is a consume comm event. If no data transferred (i.e. only produce, or only consume called) - there is no comm event.